### ISSUE TITLE
Cloud Sync issue

### ISSUE THREAD

**Issue by VladimirAkopyan** [quickbird-uk/QuickbirdUWPDashboard#8]
It appears that our much-hated and struggled cloud sync algorithm is flawed, and leaves gaps in the data
This is what graphs on the source computer look like
![image](https://cloud.githubusercontent.com/assets/12278048/18058398/19a51b64-6e0c-11e6-9a0b-bc73535fb990.png)
And this is what graphs on the destination computer look like
![image](https://cloud.githubusercontent.com/assets/12278048/18058426/31d98e9a-6e0c-11e6-97cb-a59dff4b73fd.png)
In addition, the app is currently in a consistent refuse-to-sync state, meaning that for 5 days now no new data made it to the server. However the serve is displaying successful REST requests with no errors. 
Furthermore, streaming of real-time data has failed as well, somehow. We appear to be in a 'shit's fucked' situation.

**Comment by VladimirAkopyan** [comment]
This will be fixed as part of overall API re-design

**Comment by VladimirAkopyan** [comment]
You don't need to redesign the API - I think i finally understood the problem. 
![image](https://user-images.githubusercontent.com/12278048/30783344-681a7b16-a139-11e7-8187-fba9751b3db7.png)
The data saving stopps when you log in on another computer. 
I think our code saves data block and data days even if no data actually came in. the result would be that we create a datablock even if it contains no data. OR something like that. Must investigate.

**Comment by VladimirAkopyan** [comment]
Did another test, I deleted remote copy of the app and re-installed it. It is clear now, that data is getting saved on server just fine, it is just not getting synced down to the app once it has synced for the first time (for that data-day?)
![image](https://user-images.githubusercontent.com/12278048/30786009-152049bc-a167-11e7-9882-a8bfe78bd641.png)

**Comment by VladimirAkopyan** [comment]
NOtes when it comes to cloud sync: I just had an instance where the local block emerged from the entity framework managed database with invalid foreign key for location (!!) I don;t know if this was a weired glitch or that actually happens. However I learned a lesson. Which is: 
When you do syncing, you always search entity in the local DB according to the key: in case of sensor history we have composite key which is SensorID and Time. I tried to add locationID to the local item detection, but that caused problems!
All other items need ot be overwritten, even LocationID, eve though we think it should not ordinatily change. 
I have added "OverwriteLocationID" code to see what happens. 
Closed #70 - this is basically what I was observing
